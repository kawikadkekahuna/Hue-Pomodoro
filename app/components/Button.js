import glamorous from "glamorous";
import React, { Component } from "react";

const { H1, Div, A } = glamorous;

export default function Button({ className, onClick, text, isDisabled }) {
  return (
    <Div
      height={50}
      width={150}
      display="flex"
      justifyContent="center"
      alignItems="center"
      onClick={onClick}
      border={!isDisabled ? "1px solid #FFF" : "1px solid #333"}
      borderRadius={2}
      cursor={!isDisabled ? "default" : "pointer"}
      margin="auto"
      css={
        !isDisabled
          ? {
              ":hover": {
                backgroundColor: "orange",
                transition: "background .25s ease-in-out"
              },
              ":active": {
                boxShadow: "inset 0 0 20px 0 white",
                fontWeight: "bold"
              }
            }
          : {}
      }
    >
      <A color={!isDisabled ? "#FFF" : "#333"}>
        {text}
      </A>
    </Div>
  );
}
