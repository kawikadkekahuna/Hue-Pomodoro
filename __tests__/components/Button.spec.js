import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "../../app/components/Button";
console.log(renderer);

describe("Button Component", () => {
  test("has correct default state", () => {
    const tree = shallow(<Button />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test("has correct state when disabled", () => {
    const props = {
      isDisabled: true
    };
    const tree = shallow(<Button {...props} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
