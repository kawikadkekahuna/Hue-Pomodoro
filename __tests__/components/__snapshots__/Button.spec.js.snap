// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component has correct default state 1`] = `
<glamorous.Div
  alignItems="center"
  border="1px solid #FFF"
  borderRadius={2}
  css={
    Object {
      ":active": Object {
        "boxShadow": "inset 0 0 20px 0 white",
        "fontWeight": "bold",
      },
      ":hover": Object {
        "backgroundColor": "orange",
        "transition": "background .25s ease-in-out",
      },
    }
  }
  cursor="default"
  display="flex"
  height={50}
  justifyContent="center"
  margin="auto"
  width={150}
>
  <glamorous.A
    color="#FFF"
  />
</glamorous.Div>
`;

exports[`Button Component has correct state when disabled 1`] = `
<glamorous.Div
  alignItems="center"
  border="1px solid #333"
  borderRadius={2}
  css={Object {}}
  cursor="pointer"
  display="flex"
  height={50}
  justifyContent="center"
  margin="auto"
  width={150}
>
  <glamorous.A
    color="#333"
  />
</glamorous.Div>
`;
